src/savegame_json.cpp

```diff
@@ -3044,6 +3054,34 @@ void item::io( Archive &archive )
         // Some hot/cold items from legacy saves may be inactive
         active = true;
     }
+
+    // Former comestibles that no longer need temperature tracking
+    if( !has_temperature() && ( last_temp_check != calendar::turn_zero || has_own_flag( flag_HOT ) ||
+                                has_own_flag( flag_COLD ) || has_own_flag( flag_FROZEN ) ) ) {
+        bool abort = false;
+        if( active ) {
+            if( temperature_removal_blacklist.count( type->get_id() ) != 0 ) {
+                // list of items that are safe for straightforward deactivation
+                active = false;
+            // *INDENT-OFF*
+            // } else if( type->get_id() == ... ) {
+                // special case handling: item-specific deactivation logic needed for
+                // items that may be active for reasons other than temperature tracking
+            // *INDENT-ON*
+            } else {
+                // warn and bail in unexpected cases
+                debugmsg( "Item %s is active and tracking temperature, but it should not!", type->get_id().str() );
```

Comment:
Downgrade to a DebugLog. No need to alarm the player. Also, can't you simplify these branches to just

```
if( !has_temperature() &&...) {
  if( temperature_removal_blacklist.count( type->get_id() ) == 0 ) {
     ...migrate....
  else {
     ...debuglog...
  }
}
```

Reply:
> Downgrade to a DebugLog

If we reach here, it means that either
- matched some unintended item that isn't supposed to be migrated -- need to revisit/refine the targeting criteria
- some item was changed and is no longer comestible, but forgot to think about the temperature tracking aspect -- needs to be added to the list or be given its own special case handling

Similar to [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/7762a549d992baf6d275678bafefc8884ed8007c/src/savegame_json.cpp#L3095-L3096) and [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/7762a549d992baf6d275678bafefc8884ed8007c/src/mapgen.cpp#L2893), I think its preferable for players to see and report so that the problem can be fixed, rather than just logged in a place that players don't usually pay attention to, and thus going undetected.

---

> simplify these branches

It's structured as:
```
1. deactivation logic:
   a. trivial case (in the list), just deactivate
   b. special cases, can only deactivate if no other reasons to remain active
   c. (implicit, no-op) item that isn't active but somehow does have temp. tracking vars/flags
   x. unexpected match: warn and abort
2. variable reset / flag clearing -- common to a, b and c
```
Suppose discard skeleton of 1b, and fold 1a, 1c and 2 together; it would actually look like:
```
if( !has_temperature() &&...) {
  if( active && temperature_removal_blacklist.count( type->get_id() ) == 0 ) {
     // case x
     ... warn ...
  else {
     // cases 1a&c + 2
     ... deactivate ...
     ... variable reset / flag clearing ... 
  }
}
```
Certainly the logic is sound and it will behave correctly for the time being.

However, semantics are subtly different -- whereas the list is originally intended as "items that can be trivially deactivated during migration", it becomes easy to treat this as simply "items to migrate", leading to more complacent approach of just blindly adding to the list whenever changing item from comestible to something else without sufficiently careful consideration.

Suppose in the future an item comes along that should fall into special cases (1b), this would lead to erroneous migration, breaking item processing of some other feature that needed `active == true`. I'm trying to avoid this eventuality; if you're sure you don't want these safeguards, it can be removed.

Reply:
I agree with you to keep the debug message, since it's meant to catch unintended behavior and a log message at that point will surely remain unnoticed. But as for simplifying the logic I agree with andrei8l. There is not much of a point to give it a complex structure for cases that aren't even checked. That's work that can be done once such cases are actually identified.